src/thunderbots/software/backend_output/grsim/grsim_backend.cpp

```diff
@@ -58,12 +63,48 @@ void GrSimBackend::sendGrSimPacket(const grSim_Packet& packet)
         remote_endpoint, 0, err);
 }
 
+
 void GrSimBackend::sendPrimitives(
     const std::vector<std::unique_ptr<Primitive>>& primitives)
 {
+    std::vector<grSim_Packet> grsim_packets;
+
+    MotionController::Velocity robot_velocities;
+    grSim_Packet grsim_packet;
+
+
+    // initial timestamp for bang-bang set as current time
+    static auto bangbang_timestamp = std::chrono::steady_clock::now();
+
+    std::chrono::duration<double> delta_time;
+
     // TODO: Implement this
-    // https://github.com/UBC-Thunderbots/Software/issues/21
-    grSim_Packet grsim_packet = createGrSimPacket(
-        0, YELLOW, Point(0.5, -0.1), Angle::ofRadians(-0.8), 4.0, true, false);
+    // https://github.com/UBC-Thunderbots/Software/issues/99
+    for (auto& prim : primitives)
+    {
+        MovePrimitive movePrim = dynamic_cast<MovePrimitive&>(*prim);
```

Comment:
What will happen if we send other types of primitives to this function?

Reply:
That won't work for now. I think that structure should be set up separately. There will be a bit more infrastructure I want to put in place to get stuff ready for everyone to implement the existing Primitives for grSim.

I think that's out of scope for this PR. This should focus on the Motion Controller itself.